contracts/src/v0.8/ccip/pools/SiloedLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,286 @@
+  /// @return unsiloedTokens amount of tokens available to all unsiloed chains.
+  function getliquidityForUnsiloedChains() public view returns (uint256) {
+    return s_unsiloedTokenBalance;
+  }
+
+  /// @notice Updates designations for chains on whether to mark funds as Siloed or not
+  /// @param removes A list of chain selectors to disable Siloing. Their funds will be moved into the unsiloed pool.
+  /// @param adds A list of chain selectors to enable Siloing. Adding a chain to siloing will not set the rebalancer.
+  /// The rebalancer will need to be set separately.
+  function updateSiloDesignationForChainSelectors(
+    uint64[] calldata removes,
+    ChainSiloConfigUpdate[] calldata adds
+  ) external onlyOwner {
+    for (uint256 i = 0; i < removes.length; ++i) {
+      // When a chain is removed from siloing, the funds are moved to the accounting pool shared by all unsiloed chain.
+      uint256 amountUnsiloed = s_siloedChainConfigs[removes[i]].siloedBalance;
+
+      s_unsiloedTokenBalance += amountUnsiloed;
+
+      /// @notice Removing the entire configuration will also delete the rebalancer. If the chain ever becomes siloed
+      /// again, the rebalancer will need to be set again.
+      delete s_siloedChainConfigs[removes[i]];
+
+      // Emit a removal event which includes the amount of funds moved to the general silo.
+      emit ChainUnsiloed(removes[i], amountUnsiloed);
+    }
+
+    for (uint256 i = 0; i < adds.length; ++i) {
+      /// @notice Adding a chain to siloing will not set the rebalancer
+      s_siloedChainConfigs[adds[i].remoteChainSelector].isSiloed = true;
+      s_siloedChainConfigs[adds[i].remoteChainSelector].rebalancer = adds[i].rebalancer;
+
+      // Emit a siloing event to indicate that the chain is now siloed, but that the amount of liquidity
+      // is zero until liquidity is provided.
+      emit ChainSiloed(adds[i].remoteChainSelector);
+    }
+  }
+
+  /// @notice Gets the rebalancer able to provide liquidity for a remote chain selector
+  /// @param remoteChainSelector The CCIP specific selector for the remote chain being interacted with.
+  /// @return The current liquidity manager, contract owner if the chain's funds are not siloed.
+  function getRebalancerByChain(
+    uint64 remoteChainSelector
+  ) public view returns (address) {
+    ChainSiloConfig memory remoteConfig = s_siloedChainConfigs[remoteChainSelector];
+    if (remoteConfig.isSiloed) return remoteConfig.rebalancer;
+    else return s_unsiloedChainRebalancer;
+  }
+
+  /// @notice Sets the Rebalancer address for a given remoteChainSelector.
+  /// @dev Only callable by the owner.
+  /// @param remoteChainSelector the remote chain to set.
+  /// @param newRebalancer the address allowed to add liquidity for the given siloed chain.
+  function setSiloedChainRebalancer(uint64 remoteChainSelector, address newRebalancer) external onlyOwner {
```

Comment:
How do you feel about merging this and setUnsiloedChainRebalancer and just having `0` indicate the not-siloed group?

Reply:
I thought about that. I think that the best way to handle it is to keep it in its own function and just rename the other function `setRebalancer` so that it uses the same function selector as LockReleaseTokenPool, to ease the burden on the tooling team.